lib/praxis/multipart/part.rb

```diff
@@ -211,24 +213,69 @@ def filename=(filename)
 
     def handler
       handlers = Praxis::Application.instance.handlers
-      (content_type && handlers[content_type.handler_name]) || handlers['json']
+      (content_type && handlers[content_type.handler_name]) || @default_handler
     end
 
-    def dump(**opts)
-      header_string = self.headers.collect do |name, value|
-        "#{name}: #{value}"
-      end.join("\r\n")
+    def derive_content_type(handler_name)
```

Comment:
can we merge this 'derive_content_type' with the one in ActionDefinition? they are probably the same...or almost the same...